### ISSUE TITLE
Array of nested embeddables

### ISSUE THREAD

**Issue by dotslashx** [mikro-orm/mikro-orm#1585]
**Describe the bug**
Using the new [Array of Embeddables](https://mikro-orm.io/docs/next/embeddables/#array-of-embeddables) feature, I am unable to fetch entity which contains the array of [Nested Embeddables](https://mikro-orm.io/docs/next/embeddables/#nested-embeddables).

**Stack trace**
```
SyntaxError: Arg string terminates parameters early
at new Function (<anonymous>)
at Function.createFunction (node_modules/@mikro-orm/core/utils/Utils.js:593:52)
at ObjectHydrator.getEntityHydrator (node_modules/@mikro-orm/core/hydration/ObjectHydrator.js:216:40)
at ObjectHydrator.hydrate (node_modules/@mikro-orm/core/hydration/ObjectHydrator.js:22:30)
at EntityFactory.hydrate (node_modules/@mikro-orm/core/entity/EntityFactory.js:82:27)
at EntityFactory.create (node_modules/@mikro-orm/core/entity/EntityFactory.js:36:14)
at SqlEntityManager.find (node_modules/@mikro-orm/core/EntityManager.js:99:52)
```

**To Reproduce**
```typescript
@Embeddable()
export class Address {

    @Property()
    street!: string;

    @Property()
    postalCode!: string;

    @Property()
    city!: string;

    @Property()
    country!: string;

}

@Embeddable()
export class User {

    @Embedded({ entity: () => Address, object: true })
    address!: Address;

}

@Entity()
export class Company {

    @Embedded({ entity: () => User, object: true, array: true })
    users!: User[] = [];

}
```

**Versions**

| Dependency | Version |
| - | - |
| node | 12.14.1 |
| typescript | 4.1.5 |
| mikro-orm | 4.5.0 |
| mysql | 2.2.5 |

**edit**: simplified example code

**Comment by B4nan** [comment]
Exactly this use case is tested and the tests are passing, can you provide full reproduction (ideally a test case)? How does the compiled function look like?

Here is the passing one, with array of embeddables inside another embeddable (actually two leves deep):

https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/embeddables/nested-embeddables.postgres.test.ts#L44

**Comment by dotslashx** [comment]
> edit: right, now I see the difference, its actually about arrays inside arrays, that is indeed not tested.

Not really, the issue here is about the entity embedded inside the entity which is embedded as array. I have simplified the example code

**Comment by B4nan** [comment]
There were actually two issues, one was nesting inside array of embeddables, and the other about nesting embedded arrays inside embedded arrays. Both should be addressed now in `4.5.1-dev.7` (will be available in few minutes).

**Comment by ogdans3** [comment]
This also seems to fix an error that looks like:
```
JIT runtime error: Cannot read property 'slices' of undefined

function(entity) {
  const ret = {};

  ....
}


TypeError: Cannot read property 'slices' of undefined
    at entity.slices.(anonymous function).slices.forEach (eval at createFunction (node_modules/@mikro-orm/core/utils/Utils.js:593:52), <anonymous>:41:31)
```

Where slices is an embeddable in my program. I also have embeddable arrays inside embeddable arrays.

```4.5.1-dev.8``` fixes this issue :D
